config/src/main/java/com/typesafe/config/impl/SimpleConfig.java

```diff
@@ -502,12 +510,22 @@ public TemporalAmount getTemporal(String path){
 
     @Override
     public List<ConfigMemorySize> getMemorySizeList(String path) {
-        List<Long> list = getBytesList(path);
-        List<ConfigMemorySize> builder = new ArrayList<ConfigMemorySize>();
-        for (Long v : list) {
-            builder.add(ConfigMemorySize.ofBytes(v));
+        List<ConfigMemorySize> l = new ArrayList<>();
+        List<? extends ConfigValue> list = getList(path);
+        for (ConfigValue v : list) {
+            if (v.valueType() == ConfigValueType.NUMBER) {
+                l.add(ConfigMemorySize.ofBytes(((Number) v.unwrapped()).longValue()));
+            } else if (v.valueType() == ConfigValueType.STRING) {
+                String s = (String) v.unwrapped();
+                BigInteger n = parseBytesAsBigInteger(s, v.origin(), path);
+                l.add(ConfigMemorySize.ofBytes(n));
+            } else {
+                throw new ConfigException.WrongType(v.origin(), path,
+                    "memory size string or number of bytes", v.valueType()
+                    .name());
+            }
         }
-        return builder;
+        return l;
```

Comment:
maybe getBytesList could call this one then map `toBytes` over the list?

Reply:
sure, fixed, thanks for the suggestion.